src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs

```diff
@@ -201,7 +201,7 @@ private static CSharpSyntaxContext CreateContextWorker(Workspace? workspace, Sem
 
             var isDestructorTypeContext = targetToken.IsKind(SyntaxKind.TildeToken) &&
                                             targetToken.Parent.IsKind(SyntaxKind.DestructorDeclaration) &&
-                                            targetToken.Parent.Parent.IsKind(SyntaxKind.ClassDeclaration, SyntaxKind.StructDeclaration, SyntaxKind.RecordDeclaration);
+                                            targetToken.Parent.Parent.IsKind(SyntaxKind.ClassDeclaration, SyntaxKind.StructDeclaration, SyntaxKind.RecordDeclaration, SyntaxKind.RecordStructDeclaration);
```

Comment:
I don't understand why StructDeclaration existed here. Both RecordStructDeclaration and StructDelaration seems wrong here. They cannot contain a destructor. Anything I'm missing @jcouv?

Reply:
Good catch. Destructors are not allowed in structs or record structs. I'll revert this.

Reply:
PR for fix: https://github.com/dotnet/roslyn/pull/52408
Will have to confirm with IDE folks on expected behavior since the previous behavior for structs seems intentional (had a bug attached)